research/src/pages/component-spec-template.mdx

```diff
@@ -0,0 +1,213 @@
+
+*Outline the component's structure with a diagram of its render tree. Enumerate key areas of visual composition and customization.*
+
+#### Diagram
+
+*Example*
+
+![Example Diagram](../images/spec-diagram-example.png)
+
+#### DOM Structure
+
+*Define the recommended DOM to represent the component's anatomy, as shown above. Show how important attributes (aria attributes especially) are applied to the various parts. In cases where a component nests other components, expand the full DOM structure to understand the expectation and any shortcomings in the child components' customizability.*
+
+#### Slots
+
+*Example Table*
+
+| Slot Name | Description | Fallback Content |
+| --------- | ----------- | ---------------- |
+| `thumb` | Enables replacing the component's `thumb` part with custom markup. | A default thumb implementation is provided. |
+
+#### Host Classes
+
+| Class Name | Description |
+| ---------- | ----------- |
+
+#### Slotted Content Classes
+
+| Class Name | Description |
+| ---------- | ----------- |
+
+#### CSS Parts
+
+*Example Table*
+
+| Part Name | Description |
+| --------- | ----------- |
+| `thumb` | Enables styling the `thumb` part without requiring it to be completely replaced via its slot. |
+
+---
+
+## Implementation Plan
+
+### States and Interactions
+
+*List key component states, valid state transitions, and how interactions trigger state transitions.*
+
+- *Are states changed as a result of keyboard? mouse? touch?*
+- *Are there corresponding keyboard interactions for mouse/touch interactions?*
+- *Are some events triggered on key down vs key press?*
+
+*Example Table*
+
+| State Group | States | Initial State | Description | Interaction/Transition |
+| ----------- | ------ | ------------- | ----------- | ---------------------- |
+| `disabled` | `true`/`false` | `false` | When `true`, disables the control, preventing user interaction and displaying the control with a disabled appearance. | No interaction. Controlled programmatically. |
+
+### Accessibility
+
+*Consider the accessibility of the component.*
+
+#### Keyboard Navigation and Focus
+
+- *Arrow vs tabbing key behavior*
+- *Modifier key effects (e.g. shift, ctrl, meta)*
+- *Does this component use focus delegation?*
+
+*Example Table*
+
+| Key | Description |
+| --- | ----------- |
+| Up/Right  | Increments the value of the component by calling the `increment` method. If the `shift` modifier is pressed, increases by 10x the `step` value. |
+| Down/Left | Decrements the value of the component by calling the `decrement` method. If the `shift` modifier is pressed, decreases by 10x the `step` value. |
+
+#### Form Input
+
+*Does this component integrate with form submission, form validation, etc.? Can integration be accomplish with standard APIs or are special work-arounds required?*
+
+#### Use with Assistive Technology
+
+*Are there any details about shadow dom, focus delegation, or aria attributes that need special attention?*
+
+### Globalization
```

Comment:
I would consider global readiness to be on a similar level of importance as accessibility. I don't think we should be designing any UI/UX without thinking about it. Depending on the component, there may be more or less to call out though. For example, a select may want to call out how the chevron should be positioned when in RTL mode. It may also want to point out that the chevron needs to be replaceable with a different visual.